src/screens/Home/Home.js

```diff
@@ -1,25 +1,114 @@
-      </Text>
-      <Text style={[typography.text, { color: colors.text }]}>
-        {strings.home.variant} {Config.BUILD_VARIANT}
-      </Text>
+import { IMAGE_URL, routes } from '@/controllers/routes';
+import { NAVIGATION } from '@/constants';
+import { getFav } from '@/selectors/MovieSelectors';
+import { networkService } from '@/networking';
+import { MovieController } from '@/controllers/MovieController';
+import { addIcon } from '@/assets';
+import { saveMovie } from '@/actions/MovieActions';
+
+export function Home({ navigation }) {
+  const movieController = new MovieController(networkService);
+  const [isConnected, setIsConected] = useState(false);
+  const [date, setDate] = useState(new Date());
+  //const { isLoading, data } = useQuery(['allMovies'], movieController.getMovies, {
+  //cacheTime: 0,
+  //});
+
+  const { isLoading, data, isError } = useQuery(['allMovies'], movieController.getMovies);
+
+  const dispatch = useDispatch();
+  const moviesFav = useSelector(getFav);
+
+  useEffect(() => {
+    if (!isError) {
+      setDate(new Date());
+    }
+    const unsubscribe = NetInfo.addEventListener((state) => {
+      setIsConected(state.isConnected);
+    });
+    return () => {
+      unsubscribe();
+    };
+  }, []);
+
+  const dateDiffInDays = (a, b) => {
+    const _MS_PER_DAY = 1000 * 60 * 60 * 24;
+    const utc1 = Date.UTC(a.getFullYear(), a.getMonth(), a.getDate());
+    const utc2 = Date.UTC(b.getFullYear(), b.getMonth(), b.getDate());
+    if (Math.floor((utc2 - utc1) / _MS_PER_DAY) < 2) {
+      return true;
+    } else {
+      return false;
+    }
+  };
+
+  if (isLoading) {
+    return <ActivityIndicator />;
+  }
+
+  const onMovie = ({ item }) => (
+    <View>
+      <TouchableHighlight
+        accessibilityRole="button"
+        onPress={() => navigation.navigate(NAVIGATION.details, { item: item })}
+      >
+        <View style={styles.item}>
+          <View style={styles.avatarContainer}>
+            <Image style={styles.avatar} source={{ uri: `${IMAGE_URL + item.backdropPath}` }} />
+          </View>
+          <Text style={styles.name}>{item.originalTitle}</Text>
+        </View>
+      </TouchableHighlight>
+      <TouchableWithoutFeedback accessibilityRole="button" onPress={() => addMovieFavs(item)}>
+        <Image source={addIcon} />
+      </TouchableWithoutFeedback>
     </View>
   );
+
+  const headerComponent = () => <Text style={styles.listHeadLine}>Películas</Text>;
+
+  const emptyComponent = () => <Text>No hay pelis disponibles capo</Text>;
+
+  const itemSeparator = () => <View style={styles.separator} />;
+
+  const addMovieFavs = (item) => {
+    if (moviesFav !== null) {
+      moviesFav.includes(item, 0) ? alert('ya está agregada') : dispatch(saveMovie(item));
+    } else {
+      dispatch(saveMovie(item));
+    }
+  };
+
+  // eslint-disable-next-line eqeqeq
+  if (
+    isConnected == true ||
+    (isConnected != true && data != null && dateDiffInDays(date, new Date()))
```

Comment:
Why the `dateDiffInDays` ?

Reply:
Change the function name to "isOneDayDiff" because this function takes two dates and makes the difference between them. If the difference is greater than one day, it returns false.

Reply:
Done!